infra/modules/azure_api_management/variables.tf

```diff
@@ -0,0 +1,419 @@
+}
+
+variable "resource_group_name" {
+  type        = string
+  description = "Resource group to deploy resources to"
+}
+
+variable "tier" {
+  type        = string
+  description = "Resource tiers depending on demanding workload. Allowed values are 'test', 'standard', 'premium'."
+  default     = "test"
+
+  validation {
+    condition     = contains(["test", "standard", "premium"], var.tier)
+    error_message = "Allowed values for \"tier\" are \"test\", \"standard\", or \"premium\"."
+  }
+}
+
+variable "zones" {
+  type        = list(string)
+  default     = []
+  description = "List of availability zones (supported only for Premium SKU)"
+}
+
+variable "products" {
+  type = list(object({
+    id           = string
+    display_name = string
+    description  = string
+    groups       = optional(list(string))
+    xml_policy   = optional(string)
+  }))
+  default     = []
+  description = "(Optional) List of products to create"
+}
+
+variable "apis" {
+  type = list(object({
+    name                             = string
+    api_version                      = optional(string, null) # The Version number of this API, if this API is versioned
+    oauth2_authorization_server_name = optional(string, null)
+    revision                         = optional(string, 1)
+    revision_description             = optional(string, null)
+    display_name                     = string
+    description                      = optional(string, null)
+    api_type                         = optional(string, "http") # (Optional) Type of API. Possible values are graphql, http, soap, and websocket. Defaults to http
+    path                             = string
+    protocols                        = list(string)
+    service_url                      = string
+    subscription_required            = optional(bool, false)            # Should this API require a subscription key?
+    version_set_id                   = optional(string, null)           # The ID of the Version Set which this API is associated with
+    content_format                   = optional(string, "swagger-json") # The format of the content from which the API Definition should be imported
+    content_value                    = string                           # The Content from which the API Definition should be imported
+    product_ids                      = optional(list(string), [])       # List of products this API should be associated with (Optional) if not specified, all products will be associated
+    subscription_key_names = optional(object({                          # Override the default name of the header and query string containing the subscription key header
+      header = string
+      query  = string
+    }), null)
+    xml_content = optional(string, null)
+    api_operation_policies = optional(list(object({ # List of api policy for given operation
+      operation_id = string
+      xml_content  = string
```

Comment:
what about getting policies from their own xml files to avoid editing xml into yaml?

Reply:
this module is intended to migrate the current APIMs, changing the logic too much could complicate future migration, but in any case I would say to discuss it because an improvement in the infrastructure will have to be made anyway @Krusty93 @christian-calabrese